internal/git/git.go

```diff
@@ -14,44 +15,55 @@ const (
 
 type checkoutCmd struct {
 	rukpakv1alpha1.GitSource
+	User  string
+	Token string
```

Comment:
We probably need a way for a user to provide a CA cert in case the git remote isn't trusted by the root certificate store of the unpack pod.

Does git permit authorization via client certs? If so, we might need to think about handling that?

Reply:
>Does git permit authorization via client certs? If so, we might need to think about handling that?

Yes, it does.  But I think that the personal access token is the most popular way for authorization.  Can the client certs support be a separate issue?  

>We probably need a way for a user to provide a CA cert in case the git remote isn't trusted by the root certificate store of the unpack pod.

We probably start with the option to skip the server certificate.

Reply:
Issue for the client certificate authorization: #294

Reply:
> We probably start with the option to skip the server certificate.
I added `SslNoVerify` option in the GitSource in the last commit.